### ISSUE TITLE
gcf-test.py fails on CentOS 7

### ISSUE THREAD

**Issue by tcmitchell** [GENI-NSF/geni-tools#851]
Following the instructions in README-gcf.txt on CentOS 7 results in an SSL error:

    ssl.SSLError: [Errno 1] _ssl.c:504: error:1409441B:SSL routines:SSL3_READ_BYTES:tlsv1 alert decrypt error

The full stack output of the command is:
```
$ src/gcf-test.py 
INFO:gcf-test:CH Server is https://127.0.0.1:8000/. Using keyfile /home/tmitchel/.gcf/alice-key.pem, certfile /home/tmitchel/.gcf/alice-cert.pem
INFO:gcf-test:AM Server is https://127.0.0.1:8001/. Using keyfile /home/tmitchel/.gcf/alice-key.pem, certfile /home/tmitchel/.gcf/alice-cert.pem
Traceback (most recent call last):
  File "src/gcf-test.py", line 562, in <module>
    sys.exit(main())
  File "src/gcf-test.py", line 557, in main
    exercise_am(ch_server, am_server, opts.api_version)
  File "src/gcf-test.py", line 426, in exercise_am
    slice_cred_string = ch_server.CreateSlice()
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1224, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1578, in __request
    verbose=self.__verbose
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1264, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1292, in single_request
    self.send_content(h, request_body)
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1439, in send_content
    connection.endheaders(request_body)
  File "/usr/lib64/python2.7/httplib.py", line 969, in endheaders
    self._send_output(message_body)
  File "/usr/lib64/python2.7/httplib.py", line 829, in _send_output
    self.send(msg)
  File "/usr/lib64/python2.7/httplib.py", line 791, in send
    self.connect()
  File "/home/tmitchel/geni-tools/src/gcf/omnilib/xmlrpc/client.py", line 125, in connect
    self.sock = ssl.wrap_socket(sock, self.key_file, self.cert_file, ssl_version=self.ssl_version, ciphers=self.ciphers)
  File "/usr/lib64/python2.7/ssl.py", line 387, in wrap_socket
    ciphers=ciphers)
  File "/usr/lib64/python2.7/ssl.py", line 143, in __init__
    self.do_handshake()
  File "/usr/lib64/python2.7/ssl.py", line 305, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [Errno 1] _ssl.c:504: error:1409441B:SSL routines:SSL3_READ_BYTES:tlsv1 alert decrypt error
```

**Comment by tcmitchell** [comment]
The error was seen with the current develop branch (on the road to 2.10). The same error occurs with gcf 2.9, 2.8, and 2.7, so it is not a recent change that is causing the problem.

**Comment by tcmitchell** [comment]
Using `s_client` reveals some error messages useful for internet searches. For reference, here is the command I used after invoking `gcf/gen-certs.py`:

```
openssl s_client -connect localhost:8000 -cert ~/.gcf/alice-cert.pem \
    -key ~/.gcf/alice-key.pem -CAfile ~/.gcf/ch-cert.pem
```

This results in the following:

```
CONNECTED(00000003)
depth=0 CN = geni//gpo//gcf.authority.sa
verify return:1
140619711186848:error:1409441B:SSL routines:SSL3_READ_BYTES:tlsv1 alert decrypt error:s3_pkt.c:1259:SSL alert number 51
140619711186848:error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure:s23_lib.c:184:
---
Certificate chain
 0 s:/CN=geni//gpo//gcf.authority.sa
   i:/CN=geni//gpo//gcf.authority.sa
---

[ More output removed]
```

The "SSL alert number 51" message led me to a discussion of FIPS, key sizes, and signature algorithms. There I learned that under some conditions either larger key sizes are required or specific signature algorithms are disallowed. Since CentOS is the type of OS that intends to be fairly secure, I tested a bit with the key size and signature algorithm in `src/gcf/sfa/trust/certificate.py`. Fortunately both are set in variables so it's easy to test.

Changing the `digest` on the `Certificate` class from "md5" to "sha256" fixes the issue.

Testing on other platforms is required to make sure that change doesn't break things elsewhere. With luck a patch will be forthcoming.
